docs/sphere2-core-api.txt

```diff
@@ -1541,24 +1541,35 @@ A `Server` represents a basic TCP server which listens for connections on a
 specified port.  Clients connecting to the server are placed in a backlog and
 must be explicitly accepted by game code.
 
-new Server(port[, backlog_size]); [L1]
+Server.listenOnPort(port_number[, backlog_size]); [ASYNC] [L2]
+
+    Opens `port_number` to listen for client connections.  `backlog_size`, if
+    provided, specifies the maximum number of connections that can be kept
+    waiting to be accepted before the engine will start dropping them.  The
+    default backlog size is 16.  Returns a promise for a `Server` object you
+    can use to manage client connections.
+
+new Server(port_number[, backlog_size]); [L1]
+
+    Like `Server.listenOn`, but creates the server in synchronous mode.  In
+    this mode all operations are performed synchronously, without yielding to
+    the event loop.
 
-    Constructs a new server to listen for connections on the specified port.
-    `backlog_size` specifies the size of the backlog.  If the backlog fills,
-    new connections will be dropped.
+Server#accept(); [ASYNC] [L1]
 
-Server#accept(); [L1]
+    Returns a promise for a socket representing the next client to connect to
```

Comment:
I'll agree that, as an async call, `await server.nextConnection()` is much clearer than `await server.accept()`.  "Accept" implies something immediate: accept the "guest" that's at the door and let him/her in, otherwise do nothing (in practice it could even be an error if there were a way to observe the backlog).  As opposed to "next connection" which doesn't imply anything w.r.t. timing so you might have to await it.

Reply:
`accept` has been reverted to a non-blocking synchronous call, and `acceptNext` added as the async equivalent.  The documentation was changed accordingly.